File: CppPractice/Project1/Project1/Leetcode110.cpp
#include "LeetcodeHeader.h"

struct TreeNode {
    int val;
    TreeNode* left;
    TreeNode* right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode* left, TreeNode* right) : val(x), left(left), right(right) {}
    
};
class Solution {
public:
    int depth_node(TreeNode* Node) {
        if (Node == NULL) 
        {
            return 0;
        }
        else
        {
            return max(depth_node(Node->left),depth_node(Node->right)) + 1;
        }
    
    
    }
    bool isBalanced(TreeNode* root) {
        if (!root)
        {
            return true;
        }
        if (abs(depth_node(root->left)-depth_node(root->right))<2)
        {
            return isBalanced(root->left) and isBalanced(root->right);
        }
        else
        {
            return false;
        }


    }
};

File: Pythonleetcode/Leetcode110.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def isBalanced(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """

        def height(root):
            if not root:
                return 0
            else:
                return max(height(root.right), height(root.left)) + 1

        if not root:
            return True
        else:
            return self.isBalanced(root.right) and self.isBalanced(root.left) and abs(height(root.left)-height(root.right))<=1



